src/d3a/d3a_core/live_events.py

```diff
@@ -24,6 +24,8 @@ def apply(self, area):
             # The order of the following activation calls matters:
             self.created_area.parent = area
             area.children.append(self.created_area)
+            if not area._markets.markets:
+                area._cycle_markets(_trigger_event=False)
```

Comment:
```suggestion
                area.cycle_markets(_trigger_event=False)
```

Since its no more a private method.

Reply:
@faizan2590 It is indeed a private method. Shall i make it non-private?